docs/configuration/configuration.md

```diff
@@ -117,6 +117,16 @@ Static file paths are configured as a file:// URL. `file:///var/www/static/` wil
 
 Multiple upstreams can either be configured by supplying a comma separated list to the `-upstream` parameter, supplying the parameter multiple times or provinding a list in the [config file](#config-file). When multiple upstreams are used routing to them will be based on the path they are set up with.
 
+Subdomain-based routing is possible by prepending a subdomain followed by a `|` to the upstream URL, like this:
+
+```
+test |http://127.0.0.1:8082/
+other|http://127.0.0.1:8083/
+other|http://127.0.0.1:8084/path/
```

Comment:
Actually I don't assume the cookie domain is the same, I require it. It is technically impossible to do oauth2 for a domain that is different from the cookie domain. Hence, the cookie domain is appended to every entry in the table automatically (unless it's already provided, so it's already possible to provide the entire hostname in the config, if desired).

Regarding the path suffix - I wanted to avoid changing the upstream side, since that's existing functionality. So to remain consistent, I keep the path where it was before. Doing it differently would also result in a lot more code changes.

Note that it is possible to mix old and new format. Any entry without a hostname part will be treated as a "wildcard" upstream (i.e. old functionality) and be used as a default.